### PR TITLE
Refactor common netwrok test code into a superclass

### PR DESCRIPTION
**Pull request by billonahill**
Removed a few hundred lines of duplicated code by using a common superclass.

### CODE REVIEW

File: heron/instance/tests/java/com/twitter/heron/network/AbstractNetworkTest.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+public abstract class AbstractNetworkTest {
+  static final String HOST = "127.0.0.1";
+  protected int serverPort;
+
+  private StreamManagerClient streamManagerClient;
+  private CommunicatorTester communicatorTester;
+  private GatewayMetrics gatewayMetrics;
+  private ExecutorService threadPool;
+
+  static void close(Closeable sc2) {
```
Comment (nlu90) [inline heron/instance/tests/java/com/twitter/heron/network/AbstractNetworkTest.java]: can we replace `sc2` with a meaningful name?

File: heron/instance/tests/java/com/twitter/heron/network/HandleReadTest.java [hunk 2]
```diff
@@ -251,7 +145,7 @@ private Message constructMockMessage() {
       rootId.setTaskid(0);
       dataTuple.addRoots(rootId);
 
-      String s = "";
+      String s;
```
Comment (nlu90) [inline heron/instance/tests/java/com/twitter/heron/network/HandleReadTest.java]: a nitpick, we can replace this and following `if..else` with `String s = ((i&1) == 0) ? "A" : "B";`

File: heron/instance/tests/java/com/twitter/heron/network/AbstractNetworkTest.java (hunk 1 above)
Comment (billonahill) [inline heron/instance/tests/java/com/twitter/heron/network/AbstractNetworkTest.java]: Yes, good call. Updated.

File: heron/instance/tests/java/com/twitter/heron/network/HandleReadTest.java (hunk 2 above)
Comment (billonahill) [inline heron/instance/tests/java/com/twitter/heron/network/HandleReadTest.java]: Yes, good call. Updated.

**Pull request merged**